zygrader/ui/components.py

```diff
@@ -205,36 +206,40 @@ def last(self):
         self.index = len(self.options) - 1
 
     def selected(self):
-        if self.use_dict:
-            key = list(self.options)[self.index]
-            return self.options[key]
         return self.options[self.index]
 
 
 class DatetimeSpinner(Popup):
     NO_DATE = "datetime_no_date"
 
-    def __init__(self, height, width, title, time, quickpicks, optional,
-                 include_date):
-        super().__init__(height, width, title, [], Popup.ALIGN_CENTER)
-
-        if time is None:
-            time = datetime.datetime.now()
-        self.time = time
-
-        if quickpicks:
-            quickpicks = sorted(quickpicks)
-        self.quickpicks = quickpicks
+    def __init__(self, height, width, title):
+        super().__init__(height, width, title, [])
+        super().set_align(Popup.ALIGN_CENTER)
 
-        self.optional = optional
-        self.include_date = include_date
+        self.time = datetime.datetime.now()
+        self.quickpicks = None
+        self.optional = False
+        self.include_date = True
 
         self.__init_fields()
         self.__init_format_str()
         self.__init_input_str()
 
         curses.curs_set(0)
 
+    def set_quickpicks(self, quickpicks):
+        self.quickpicks = quickpicks
+
+    def set_time(self, time):
+        self.time = time
+
+    def set_optional(self, optional):
+        self.optional = optional
+        self.__init_fields()
+
+    def set_include_date(self, include_date):
+        self.include_date = include_date
+
```

Comment:
Including (or not) the date means the fields and format string need to update as well.
```suggestion
        self.__init_fields()
        self.__init_format_str()

```